stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
```

Comment:
I should add that the new_ostruct_member rubyspecs all expect a @table instance variable as well. By implementing the self.$$table advice above, that will break that. ostruct seems to be designed with inheritance in mind given this and the new_ostruct, etc. being protected. One could make the argument that the rubyspecs are mucking with the internals of what they are testing in this case, but I could see it both ways.